New History List Using GCard

itisAliRH opened the pull request:
This pull request reimplements the History List interface using the GCard, similar to the Workflow List (#16607). The refactoring aims to modernise the UI while maintaining functionality, resulting in a more consistent design language across the Galaxy.

## Key Features

- **Unified History View Components**: Redesigned history list using a consistent component, `HistoryCard`, with GCard, making maintenance and future enhancements easier
- **Responsive Design**: Implemented responsive layouts that adapt to different screen sizes and device types
- **Use modals**: Old browser alerts for actions like deleting are replaced with confirmation modals.
- **New Bulk Action**: Adding new tags to selected histories
- **Highlight Published Histories**

### My Histories

|Before|After|
|---|---|
|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/aebc09d0-35cc-4106-b637-fa87826b023b" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/59cb1d3d-18ed-4b67-a4a4-71337e3d9add" />|


### My Histories Grid View

<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/9efda0f3-5ff3-49bf-be11-54f7acfb3fd1" />


### Bulk Selection

|Before|After|
|---|---|
|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/ab5ac63b-2990-4b80-9ab2-11782586c757" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/72235be3-9cae-4a3c-8bcc-b39123a5f1f8" />|

### Histories Shared with Me

|Before|After|
|---|---|
|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/4b53d9ec-89e6-4373-9e79-cf93e196461c" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/f72f6295-5e0c-4d6f-900d-32dda8e4d3ac" />|

### Public Histories

|Before|After|
|---|---|
|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/e8d895ea-885e-4889-87d0-d5c33245aa42" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/2d1951a7-afe4-40e7-a4f4-7d936743cd1e" />|

### Archived Histories

|Before|After|
|---|---|
|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/7dce3aeb-959d-4f9c-af62-aea45fcad286" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/a983d555-7f70-415a-8f2c-a17f703e06ba" />|

### History Datasets and Size

|Before|After|
|---|---|
|<img width="520" height="89" alt="image" src="https://github.com/user-attachments/assets/afe118c3-9250-4d6e-b3aa-fe681ca408c6" />|<img width="558" height="71" alt="image" src="https://github.com/user-attachments/assets/6cd61b81-28ac-4c15-b179-d2290d000101" />|

### Delete a history

|Before|After|
|---|---|
|<img width="1798" height="1040" alt="image" src="https://github.com/user-attachments/assets/3dea8192-e262-420b-af2c-81506d503ea5" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/1159048e-8435-4c06-affc-fdf5b72e5697" />|

### Permanently delete a history

|Before|After|
|---|---|
|<img width="1800" height="1040" alt="image" src="https://github.com/user-attachments/assets/cef3192a-1203-46ca-869e-2f5b42d057a0" />|<img width="1800" height="1126" alt="image" src="https://github.com/user-attachments/assets/fd780efe-ad84-4fc1-a53b-4f34022ecb20" />|

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

itisAliRH committed:
Adds HistoryDatasetsBadge component for history storage overview

Introduces a new component to display a badge summarizing history storage details, including item count, size, and state statistics.
Enhances user experience by providing a quick visual overview and navigation to the history storage overview page.
Includes responsive styling.

itisAliRH committed:
Adds HistoryCardList component for managing all types of histories

Introduces a new component to display and manage all types of histories with actions like view, delete, restore, import, and unarchive using the GCard.
Supports grid and list views, tag editing, and badges for metadata display.
Enhances user interaction with confirm dialogs, toast notifications, and computed properties for ownership and permissions.

itisAliRH committed:
Add dynamic filtering for history list views

Introduces a reusable filtering system to support dynamic filters for history lists.
Provides configurable filters like name, tags, and boolean flags, tailored to user-specific or shared views based on the active list type.

itisAliRH committed:
Adds HistoryList component for managing histories

Introduces a new component to manage and display histories, including features like filtering, sorting, pagination, and bulk actions such as delete, restore, and tag addition.
Supports multiple views (my, shared, published, archived) and advanced filtering options.
Improves user experience by adding structured navigation, placeholders, and error handling for invalid filters.

itisAliRH committed:
Replace GridHistory with HistoryList in router paths

Replaces the GridHistory component with the HistoryList component across all history lists router paths.

itisAliRH committed:
Removes GridHistory component

itisAliRH committed:
Removes unused grid configuration for my histories

itisAliRH committed:
Removes unused grid configuration for shared histories

itisAliRH committed:
Removes unused grid configuration for published histories

itisAliRH committed:
Removes unused GridDatasets component and related logic

Eliminates the GridDatasets component and its associated code, including imports, template usage, and styles, to streamline the codebase.

itisAliRH committed:
Removes unused archived history management component

itisAliRH committed:
Removes unused ArchivedHistoryCard component

itisAliRH committed:
Updates history selectors to use card-based structure

Replaces grid-based selectors with card-based ones for shared and
regular histories to align with the updated UI structure.
Enhances readability and consistency by introducing specific selectors for
history cards and their titles.

itisAliRH committed:
Updates history navigation element references

Replaces outdated element references in history navigation methods
to align with updated UI component naming conventions.

itisAliRH committed:
Refactors history retrieval methods

Replaces generic history retrieval methods with more specific ones.
Updates method names and logic to better reflect their purpose for history titles.

itisAliRH committed:
Updates shared history element selectors in test

Replaces outdated selectors for shared history elements to align with updated UI structure.

itisAliRH committed:
Adds type for updating history payload

Export new type to represent the payload for updating history objects.

itisAliRH committed:
Refactors history update method for better parameter clarity

Revises the `updateHistory` method signature to accept the history ID and update payload as separate parameters, improving readability and consistency across usage.
Updates all call sites to align with the modified method signature.

itisAliRH committed:
Removes unused event handler from HistoryDetails component

Eliminates the `@update:history` event handler from the HistoryDetails component as it is no longer in use.
Simplifies the component's props and improves readability without affecting functionality.

itisAliRH committed:
Adds types for categorized history entries

Introduces new types to represent categorized history entries, including `MyHistory`, `SharedHistory`, `PublishedHistory`, and `AnyHistoryEntry`.
These types improve type safety and clarity when handling different kinds of history data.
Also adds support for the `ArchivedHistorySummary` type, extending compatibility with archived history entries.

itisAliRH committed:
Add API method for retrieving history counts

Introduces a new function to fetch detailed history statistics, including size and content states, using the Galaxy API.
Enhances functionality by enabling retrieval of targeted data for histories.
Handles errors with consistent utility for rethrowing simple errors.

itisAliRH committed:
Refactors history badge to use reactive state and API integration

Replaces prop-based state management with reactive `ref` variables to improve reactivity and maintainability.
Introduces API call to fetch history counts on component mount, handling errors gracefully with a toast notification.
Simplifies template bindings by directly referencing reactive state variables instead of props.

itisAliRH committed:
Adds support for fetching and filtering history data

Introduces utility functions to fetch various types of history data, including my, shared, published, and archived histories.
Each function supports filtering options such as pagination, search, and sorting by specific criteria.
Enhances the API integration to improve flexibility and usability for history management.

itisAliRH committed:
Refactors history loading logic for modularity and clarity

Replaces monolithic API calls with modularized functions for loading different history types (e.g., "my", "shared", "published", "archived"), improving maintainability and readability.
Streamlines error handling and removes unused imports to reduce
complexity.
Updates the data structure for loaded histories to align with the new modular approach.

itisAliRH committed:
Refactors history handling with stricter type definitions

Replaces generic `AnyHistory` type with more specific history types to improve type safety and clarity.
Adds utility functions to distinguish between history types and updates logic accordingly.
Enhances maintainability by aligning type-based behavior with application requirements.

itisAliRH committed:
Adds unique IDs to history card elements

Introduces an `id` attribute to history card elements for better identification and accessibility. This change improves DOM structure and supports use cases like testing or element targeting.

itisAliRH committed:
Replaces toast composable with Toast object for consistency

Updates all toast notification calls to use the `Toast` object instead of the `toast` composable.
Enhances consistency in the toast API usage and improves error message formatting by including detailed error information.
No functional changes introduced.

itisAliRH committed:
Simplifies HistoryDatasetsBadge props

Removes redundant props from the HistoryDatasetsBadge component to streamline its interface and improve maintainability.

itisAliRH committed:
Adds loading spinner and improves badge rendering logic

Introduces a loading spinner to indicate data fetching status, enhancing user feedback.
Refactors badge rendering logic to display content conditionally based on loading state.

itisAliRH committed:
Adds unique IDs to bulk delete and restore confirmation dialogs

Introduces unique `id` attributes to the confirmation dialogs for bulk delete and restore actions.
No functional changes to the dialog behavior are introduced.

itisAliRH committed:
Updates selectors for history navigation UI

Refines CSS selectors and identifiers for history-list related UI elements to align with updated structure and naming conventions.
Adds support for bulk actions (delete/restore) in history management, improving user operations and consistency across components.

itisAliRH committed:
Adds methods for interacting with history cards in lists

Introduces utility methods to interact with history cards, including selecting specific card operations, retrieving card titles, and toggling card selections.
Enhances functionality and modularity by replacing grid-based selection logic with card-based handling.
Improves testability and usability in scenarios involving history card interactions.

itisAliRH committed:
Refactors history list tests to use card-based selectors

Replaces grid-based operations and assertions with card-based selectors in Selenium tests for histories.
Updates methods and locators to align with a UI change from grid layout to card-based list.

itisAliRH committed:
Improve published histories sorting and tag handling

Updates selectors and methods for sorting published histories by name and update time to match new UX changes.
Ensures proper rendering delays with added sleep calls for UI transitions.
Refines tag selection logic and history retrieval.

itisAliRH committed:
Updates history title retrieval

Replaces the method for retrieving history titles with a more specific and likely optimized approach.

itisAliRH committed:
Improves history import verification

Replaces grid-based history name verification with a more robust  check using the history card list.
Ensures tests are aligned with current UI behavior.

itisAliRH committed:
Ensures list refreshes after history changes

Updates event emission to consistently refresh the list after delete, restore, and unarchive actions, improving UI accuracy.

itisAliRH committed:
Improves history card action selection with extra actions

Enhances selection of history card operations by allowing interaction with extra actions menus and using action chains for more reliable click events.

itisAliRH committed:
Adds selector for delete history confirmation button

itisAliRH committed:
Refactors history list tests to use card-based selectors

Updates Selenium tests to interact with history cards via more robust, card-specific selectors instead of grid-based operations.
Improves reliability and alignment with UI changes by replacing alert handling with explicit confirmation button actions and renaming assertions for clarity.

itisAliRH committed:
Fixes search filter assertion and improves UX waits

Updates the test to assert the correct tag-based search filter value after clicking a tag.

itisAliRH committed:
Renames method to clarify sorting in history list assertions

Updates test method and its usages to specify that the assertion checks for sorted histories.

itisAliRH committed:
Adds utility functions to classify user history entries

Introduces helpers to determine if a history entry is owned, shared, published, or archived by the current user.
Improves code clarity and enables more readable checks for user-specific history operations.

itisAliRH committed:
Adds composable for history card actions and dialogs

Introduces a reusable composable to centralize logic for history card actions, including deletion, restoration, import, and UI configuration.
Streamlines dialog confirmations, user feedback, and permission checks to improve maintainability and consistency across the history management interface.

itisAliRH committed:
Adds composable for history card status indicators

Introduces a dedicated composable to encapsulate the logic for displaying status indicators on history cards, improving code reuse and maintainability.
Enables context-aware labels and actions for published, deleted, and purged histories, supporting better UI feedback and interaction.

itisAliRH committed:
Adds composable for dynamic history card badges

Introduces a reusable composable to generate context-aware badges for history cards, improving modularity and enabling clearer user distinction for shared, published, and snapshot histories.
Enhances user filtering and discoverability through badge actions.

itisAliRH committed:
Add reusable component for displaying history cards

Introduces a new UI component to encapsulate history card display and actions, streamlining presentation, selection, and interaction logic for history entries.
Improves maintainability and consistency across views that render histories.

itisAliRH committed:
Refactors history card list to use HistoryCard component

Simplifies the history card list by delegating per-history logic and UI rendering to a new, dedicated component.
Reduces complexity, improves maintainability, and enables better separation of concerns between the list and individual card behaviors.

itisAliRH committed:
Enhance history entry types and add detailed documentation for fetching and ownership checks

itisAliRH committed:
Enhance HistoryList component with detailed documentation and type definitions for improved clarity and maintainability

itisAliRH committed:
Enhance HistoryCardList component with detailed documentation for props and emitted events

itisAliRH committed:
Enhance HistoryCard component with detailed documentation for props, events, and methods

itisAliRH committed:
Enhance HistoryDatasetsBadge component with detailed documentation for props and functionality

itisAliRH committed:
Enhance useHistoryCardActions composable with detailed documentation for functions and parameters

itisAliRH committed:
Enhance useHistoryCardBadges composable with detailed documentation for parameters and internal functions

itisAliRH committed:
Enhance useHistoryCardIndicators composable with detailed documentation for parameters and internal functions

itisAliRH committed:
Add wait for UX render before inserting a tag in test_tags method

itisAliRH committed:
Adds bulk purge action for selected histories

Introduces the ability to permanently purge multiple selected histories in addition to bulk delete and restore actions.
Updates selection logic and UI to support purging, improving user control over managing deleted histories.

itisAliRH committed:
Add wait for UX render in test_tags method before asserting history sort order

itisAliRH committed:
Refactor bulk delete confirmation to handle purge option and update button click handler

jmchilton reviewed (approved):
All your histories are all green datasets. Digging into the code here (I'll look at Galaxy code but I would never run it), it looks like you modernized all that code but I thought it might be good to highlight in a screenshot. usegalaxy.org looks like this for me:

<img width="310" height="335" alt="Screenshot 2025-08-15 at 2 30 51 PM" src="https://github.com/user-attachments/assets/30714e67-b9da-4919-9451-bda9b5cae822" />

I feel like these colors used to be more descriptive (reds and yellows and such) but that isn't your fault - even a popover to describe what these states mean would be great but that isn't your responsibility here just yelling at clouds.

![old man yells at cloud meme](https://static.wikia.nocookie.net/memepediadankmemes/images/0/01/297.jpg/revision/latest?cb=20180908193511)

itisAliRH committed:
Updates data-description for import history button

Aligns the import button's data-description attribute with its actual location in the UI for improved selector accuracy and test reliability.

ahmedhamidawan reviewed (approved):
Ran this locally, looks beautiful! Thank you!

___

One concern though. I have opened a PR for an unrelated bug [here](#20800) but it made me notice, we have two options here which can maybe become one:

<img width="820" height="163" alt="image" src="https://github.com/user-attachments/assets/8de582b6-e644-4420-a6cb-95a3d8dfb7cd" />

If you open the `Share` view, the 2nd tab there does exactly what the `Change Permissions` view here does. Maybe we can replicate what we have in the history options menu here too, and have a singular share and manage access button?:

<img width="325" height="170" alt="image" src="https://github.com/user-attachments/assets/44a44681-4a9f-4c2a-82f3-3113cbb53c26" />

itisAliRH committed:
Consolidates sharing and access actions into one button

Simplifies the history card UI by merging sharing, publishing, and permission management into a single "Share & Manage Access" action for consistency.
Updates related test selectors to match the new action identifier.
